### PR TITLE
Add spinner to edit channel dialog

### PR DESCRIPTION
**Pull request by akalin-keybase**
Clean up team getters a bit also.

### CODE REVIEW

File: shared/chat/manage-channels/edit-channel-container.js [hunk 1]
```diff
@@ -10,26 +10,36 @@ import {anyWaiting} from '../../constants/waiting'
...
   const conversationIDKey = routeProps.get('conversationIDKey')
-  const teamname =
-    routeProps.get('teamname') || Constants.getTeamNameFromConvID(state, conversationIDKey) || ''
-  const waitingForSave = anyWaiting(
+  if (!conversationIDKey) {
+    throw new Error('conversationIDKey unexpectedly empty')
+  }
+
+  const teamname = routeProps.get('teamname')
+  if (!teamname) {
```
Comment (akalin-keybase) [inline shared/chat/manage-channels/edit-channel-container.js]: @chrisnojima mentioned that we want to reduce the info passed through routeprops, but it seems that teamname is necessary here -- if we don't have it, and it's not already in the teams data, we don't know which team to refresh to get it!

File: shared/constants/teams.js [hunk 2]
```diff
@@ -170,9 +170,6 @@ const userIsActiveInTeamHelper = (
   return member && member.active
 }
 
-const getConvIdsFromTeamName = (state: TypedState, teamname: string): I.Set<ChatTypes.ConversationIDKey> =>
```
Comment (akalin-keybase) [inline shared/constants/teams.js]: redundant with getTeamConvIDs

File: shared/chat/manage-channels/edit-channel-container.js [hunk 3]
```diff
@@ -10,26 +10,38 @@ import {anyWaiting} from '../../constants/waiting'
...
+  const channelInfo = Constants.getChannelInfoFromConvID(state, conversationIDKey)
+  const _needsLoad = !channelInfo
+
+  const waitingForGetInfo = _needsLoad || anyWaiting(state, `getChannels:${teamname}`)
+  const waitingForUpdate = anyWaiting(
     state,
     `updateTopic:${conversationIDKey}`,
-    `updateChannelName:${conversationIDKey}`,
-    `getChannels:${teamname}`
+    `updateChannelName:${conversationIDKey}`
```
Comment (buoyad) [inline shared/chat/manage-channels/edit-channel-container.js]: 🐑 can these be constants?

File: shared/chat/manage-channels/edit-channel-container.js [hunk 4]
```diff
@@ -10,26 +10,38 @@ import {anyWaiting} from '../../constants/waiting'
...
+  if (!conversationIDKey) {
+    throw new Error('conversationIDKey unexpectedly empty')
+  }
+
+  const teamname = routeProps.get('teamname')
+  if (!teamname) {
+    throw new Error('teamname unexpectedly empty')
+  }
+
+  const channelInfo = Constants.getChannelInfoFromConvID(state, conversationIDKey)
```
Comment (buoyad) [inline shared/chat/manage-channels/edit-channel-container.js]: This data is already in `state.chat2.metaMap` after you've loaded this conversation. Right now `_getChannels` in actions/teams gets all the chat inbox items for the channels and throws away everything that is not relevant to this screen. I agree that having the duplicated data in the store isn't that nice, maybe a better flow is to have `_getChannels` delegate to a chat action to get the trusted metas for the channels. Then the channel names & descriptions can live in the chat `metaMap` and `getChannelInfoFromConvID` can look there for the data. But :sheep: if you don't want to do that here it's fine.

File: shared/chat/manage-channels/edit-channel.js [hunk 5]
```diff
@@ -30,14 +40,20 @@ const EditChannelBare = (props: Props & TextState) => (
     <Text type="BodySmallSemibold" style={{color: globalColors.darkBlue, marginTop: globalMargins.xtiny}}>
       {props.teamname}
     </Text>
-    <Text type="Header" style={{marginBottom: globalMargins.tiny, marginTop: globalMargins.tiny}}>
-      Edit #{props.channelName}
-    </Text>
+    {props.waitingForGetInfo ? (
```
Comment (buoyad) [inline shared/chat/manage-channels/edit-channel.js]: Hm, I'm not seeing these spinners. I think `waitingForGetInfo` is missing in `mergeProps`

File: shared/chat/manage-channels/edit-channel-container.js (hunk 3 above)
Comment (akalin-keybase) [reply shared/chat/manage-channels/edit-channel-container.js]: Done!

File: shared/chat/manage-channels/edit-channel.js (hunk 5 above)
Comment (akalin-keybase) [reply shared/chat/manage-channels/edit-channel.js]: D'oh! Added types and fixed

File: shared/chat/manage-channels/edit-channel-container.js (hunk 4 above)
Comment (akalin-keybase) [reply shared/chat/manage-channels/edit-channel-container.js]: I took a look and yeah, maybe having `_getChannels` write its info to the chat `metaMap` would be a better flow. I'll see if I can do that after my follow-up PR after this one.

**Comment by akalin-keybase** [comment]
PTAL!

**Pull request merged**